lib/fastlane/plugin/wpmreleasetoolkit/models/locale_map.rb

```diff
@@ -0,0 +1,120 @@
+    #        If no `value_sym` is specified (or `:itself`), the returned `Hash`'s values will be `LocaleMap` objects in full (instead of just a `String` locale code)
+    #
+    # @return a Hash whose keys and values are derived from the corresponding attributes of each `LocaleMap` object of this `LocalesMap`
+    #
+    # @example `to_h(:onesky, :ios)` will return a `Hash` whose keys are the locale code for :onesky and values are the corresponding locale code for :ios
+    # @example `to_h(:ios)` will return a `Hash` whose keys are the locale code for :ios and values are the corresponding `LocaleMap` objects
+    # @note Any key or value that is nil (i.e. any LocaleMap which does not have a locale code defined for either that `key_sym` or `value_sym`) will be filtered out
+    #
+    def to_h(key_sym, value_sym = :itself)
+      @locales.to_h do |lm|
+        key = lm.send(key_sym)
+        [key, key.nil? ? nil : lm.send(value_sym)]
+      end.compact
+    end
+
+    # Returns an Array of locale codes for the `key_sym` service/context
+    #
+    # @param [Symbol] key_sym The symbol to indicate which locale code type to use for the values of the produced Array. Should be one of `LocaleMap.members`.
+    #                 If no `key_sym` is specified (or `:to_h`), will return an `Array` of `Hash` representations of the `LocaleMap` objects
+    # @return an Array whose values are derived from the corresponding attribute of each `LocaleMap` object of this `LocalesMap`
+    #
+    # @example `to_a(:ios)` will return an `Array` whose items are the locale code for `:ios`
+    # @example `to_a(:to_h)` will return an `Array` whose items are the `Hash` representation of each `LocaleMap` object
+    # @note `to_a` (aka `to_a(:itself)`) will return the same as `locales`
+    # @note Any value that is nil (i.e. any LocaleMap which does not have a locale code defined for that `key_sym`) will be filtered out
+    #
+    def to_a(key_sym = :to_h)
+      @locales.map { |lm| lm.send(key_sym) }.compact
+    end
+
+    # Returns a new `LocalesMap` object only keeping the `LocaleMap` objects for which the `block` returns true
+    # @example `locales_map.select { |l| ['fr', 'es'].include?(l.ios) }
+    def select(&block)
+      LocalesMap.new(@locales.select(block))
+    end
+
+    def reject(&block)
+      LocalesMap.new(@locales.reject(block))
+    end
+
+    def each(&block)
+      @locales.each(block)
+    end
+
+    # Convert a locale code from one service/context to another
+    #
+    # @param [String] locale The locale code to convert
+    # @param [Symbol] from The service/context to convert the locale from. Should be one of `LocaleMap.members`
+    # @param [Symbol] to The service/context to convert the locale to. Should be one of `LocaleMap.members`
+    #
+    def convert(locale:, from:, to:)
+      @locales.find { |lm| lm.send(from) == locale }&.send(to)
+    end
+
+    # The set of common locale codes mappings known to us
+    #
+    def self.default
```

Comment:
💡  API idea: also provide a `def self.only(options)` method which will expect `options` to be a `Hash` with a single key which must be one of the `LocaleMap.members`, and a value being an `Array<String>`, so that we can call `LocalesMap.only(onesky: %w[fr pt-BR en es it])` to get a subset of the known locales.

We can then implement `def self.mag16` by just calling that `self.only` method with the 16 expected locales, and additionally make it easy for client apps to provide just the custom list of locales they support.